src/Microsoft.Data/PrimitiveColumnComputations.cs

```diff
@@ -0,0 +1,1924 @@
+            for (int bb = 0 ; bb < column.Buffers.Count; bb++)
+            {
+                var buffer = column.Buffers[bb];
+                for (int i = 0; i < buffer.Length; i++)
+                {
+                    ret = (double)(buffer.Span[i] * ret);
+                    buffer.Span[i] = ret;
+                }
+            }
+        }
+
+        public void CumulativeSum(PrimitiveColumnContainer<double> column)
+        {
+            var ret = (double)0;
+            for (int bb = 0 ; bb < column.Buffers.Count; bb++)
+            {
+                var buffer = column.Buffers[bb];
+                for (int i = 0; i < buffer.Length; i++)
+                {
+                    ret = (double)(buffer.Span[i] + ret);
+                    buffer.Span[i] = ret;
+                }
+            }
+        }
+
+        public void Max(PrimitiveColumnContainer<double> column, out double ret)
+        {
+            ret = column.Buffers[0].Span[0];
+            for (int bb = 0 ; bb < column.Buffers.Count; bb++)
+            {
+                var buffer = column.Buffers[bb];
+                for (int i = 0; i < buffer.Length; i++)
+                {
+                    ret = (double)(Math.Max(buffer.Span[i], ret));
+                }
+            }
+        }
+
+        public void Min(PrimitiveColumnContainer<double> column, out double ret)
+        {
+            ret = column.Buffers[0].Span[0];
+            for (int bb = 0 ; bb < column.Buffers.Count; bb++)
+            {
+                var buffer = column.Buffers[bb];
+                for (int i = 0; i < buffer.Length; i++)
+                {
+                    ret = (double)(Math.Min(buffer.Span[i], ret));
+                }
+            }
+        }
+
+        public void Product(PrimitiveColumnContainer<double> column, out double ret)
+        {
+            ret = (double)1;
+            for (int bb = 0 ; bb < column.Buffers.Count; bb++)
+            {
+                var buffer = column.Buffers[bb];
+                for (int i = 0; i < buffer.Length; i++)
+                {
+                    ret = (double)(buffer.Span[i] * ret);
+                }
+            }
+        }
+
+        public void Sum(PrimitiveColumnContainer<double> column, out double ret)
+        {
+            ret = (double)0;
+            for (int bb = 0 ; bb < column.Buffers.Count; bb++)
+            {
+                var buffer = column.Buffers[bb];
+                for (int i = 0; i < buffer.Length; i++)
+                {
+                    ret = (double)(buffer.Span[i] + ret);
+                }
+            }
+        }
+
+        public void Round(PrimitiveColumnContainer<double> column)
+        {
+            for (int b = 0; b < column.Buffers.Count; b++)
+            {
+                var buffer = column.Buffers[b];
+                for (int i = 0; i < buffer.Length; i++)
+                {
+                    buffer.Span[i] = (double)(Math.Round((decimal)buffer.Span[i]));
```

Comment:
why the cast to `decimal` here? Math.Round can take a `double`.

Reply:
Math.Round(int) was ambiguous since Round takes both decimal and double. This file is template generated and I wanted to keep the template file simple, so I just cast every type to decimal.

Reply:
Can you file an issue for this?  I don't think we should be doing the cast to decimal, as there could be some precision/truncation issues/changes.